types/fh-sync.d.ts

```diff
@@ -0,0 +1,121 @@
+    meta_data: any;
+  }
+
+  namespace Sync {
+    function connect(mongoDBConnectionUrl: string, mongoDBConnectionOption: any, redisUrl: string, cb: StandardCb<void>)
+
+    function init(dataset_id: string, options: SyncInitOptions, callback: StandardCb<void>): void;
+    function invoke(dataset_id: string, options: any, callback: () => void): void;
+
+    function stop(dataset_id: string, onStop: () => void): void;
+    function stopAll(onstop: StandardCb<string[]>): void;
+
+    function handleList(dataset_id: string, onList: (dataset_id: string, params: any, meta_data: any, callback: StandardCb<any>) => void): void;
+    function globalHandleList(onList: (dataset_id: string, params: any, meta_data: any, callback: StandardCb<any>) => void): void;
+
+    function handleCreate(dataset_id: string, onCreate: (dataset_id: string, data: any, meta_data: any, callback: StandardCb<any>) => void): void;
+    function globalHandleCreate(onCreate: (dataset_id: string, params: any, meta_data: any, callback: StandardCb<any>) => void): void;
+
+    function handleRead(dataset_id: string, onRead: (dataset_id: string, uid: any, meta_data: any, callback: StandardCb<any>) => void): void;
+    function globalHandleRead(onRead: (dataset_id: string, uid: string, meta_data: any, callback: StandardCb<any>) => void): void;
+
+    function handleUpdate(dataset_id: string, onUpdate: (dataset_id: string, uid: string, data: any, meta_data: any, callback: StandardCb<any>) => void): void;
+    function globalHandleUpdate(onCreate: (dataset_id: string, uid: string, data: any, meta_data: any, callback: StandardCb<any>) => void): void;
+
+    function handleDelete(dataset_id: string, onCreate: (dataset_id: string, uid: string, meta_data: any, callback: StandardCb<any>) => void): void;
+    function globalHandleDelete(onCreate: (dataset_id: string, uid: string, meta_data: any, callback: StandardCb<any>) => void): void;
+
+    function handleCollision(dataset_id: string, onCollision: (dataset_id: string, hash: string, timestamp: any, uid: string, pre: any, post: any, meta_data: any, callback: StandardCb<any>) => void): void;
+    function globalHandleCollision(onCollision: (dataset_id: string, hash: string, timestamp: Date, uid: string, pre: any, post: any, meta_data: any, callback: StandardCb<any>) => void): void;
+
+    function listCollisions(dataset_id: string, onList: (dataset_id: string, meta_data: any, callback: StandardCb<{ [hash: string]: any }>) => void): void;
+    function globalListCollisions(onList: (dataset_id: string, meta_data: any, callback: StandardCb<{ [hash: string]: any }>) => void): void;
+
+    function removeCollision(dataset_id: string, onRemove: (dataset_id: string, collision_hash: string, meta_data: any, callback: StandardCb<any>) => void): void;
+
+    function interceptRequest(dataset_id: string, onIntercept: (dataset_id: string, interceptor_params: SyncInterceptParams, callback: NoRespCb) => void): void;
+    function interceptResponse(): void;
```

Comment:
Shouldn't this be something like `(datasetId: string, fn: (datasetId: string, queryParams:any, cb: (err: Error | string) => void))`?